.ci/install_kata_kernel.sh

```diff
@@ -14,133 +14,69 @@ set -o errtrace
-		sudo -E ln -sf "${vmlinux_experimental_path}" "${vmlinux_symlink}"
-
-		local vmlinuz_symlink="${kernel_dir}/vmlinuz.container"
-		local vmlinuz_experimental_path=$(readlink -f "${kernel_dir}/vmlinuz-experimental.container")
-		[ -e "$vmlinuz_experimental_path" ] || die "Not found experimental kernel installed '${vmlinuz_experimental_path}'"
-		info "Installing ${vmlinuz_experimental_path} and symlink ${vmlinuz_symlink}"
-		sudo -E ln -sf "${vmlinuz_experimental_path}" "${vmlinuz_symlink}"
-		popd >> /dev/null
-	else
-		# Always build and install the kernel version found locally
-		info "Install kernel from sources"
-		pushd "${tmp_dir}" >> /dev/null
-		"${kernel_repo_dir}/kernel/build-kernel.sh" -v "${kernel_version}" "setup"
-		"${kernel_repo_dir}/kernel/build-kernel.sh" -v "${kernel_version}" "build"
-		sudo -E PATH="$PATH" "${kernel_repo_dir}/kernel/build-kernel.sh" -v "${kernel_version}" "install"
-		popd >> /dev/null
-	fi
+	# Always build and install the kernel version found locally
+	info "Install kernel from sources"
+	pushd "${tmp_dir}" >> /dev/null
+	"${kernel_packaging_dir}/build-kernel.sh" -v "${kernel_version}" "setup"
+	"${kernel_packaging_dir}/build-kernel.sh" -v "${kernel_version}" "build"
+	sudo -E PATH="$PATH" "${kernel_packaging_dir}/build-kernel.sh" -v "${kernel_version}" "install"
+	popd >> /dev/null
 }
 
 # $1 kernel_binary: binary to install could be vmlinux or vmlinuz
 install_cached_kernel(){
 	local kernel_binary=${1:-}
 	[ -z "${kernel_binary}" ] && die "empty binary format"
-	info "Installing ${kernel_binary}"
+	info "Attempting to download cached ${kernel_binary}"
 	sudo mkdir -p "${kernel_dir}"
 	local kernel_binary_name="${kernel_binary}-${cached_kernel_version}"
 	local kernel_binary_path="${kernel_dir}/${kernel_binary_name}"
-	if [ "${experimental_kernel}" == "true" ]; then
-		sudo -E curl -fL --progress-bar "${experimental_latest_build_url}/${kernel_binary_name}" -o "${kernel_binary_path}" || return 1
-	else
-		sudo -E curl -fL --progress-bar "${latest_build_url}/${kernel_binary_name}" -o "${kernel_binary_path}" || return 1
-	fi
+	sudo -E curl -fL --progress-bar "${latest_build_url}/${kernel_binary_name}" -o "${kernel_binary_path}" || return 1
 	kernel_symlink="${kernel_dir}/${kernel_binary}.container"
-	info "Installing ${kernel_binary_path} and symlink ${kernel_symlink}"
 	sudo -E ln -sf "${kernel_binary_path}" "${kernel_symlink}"
 }
 
 install_prebuilt_kernel() {
-	info "Install pre-built kernel version"
-
 	for k in "vmlinux" "vmlinuz"; do
 		install_cached_kernel "${k}" || return 1
 	done
 
 	pushd "${kernel_dir}" >/dev/null
-	info "Verify download checksum"
-	if [ "${experimental_kernel}" == "true" ]; then
-		sudo -E curl -fsOL "${experimental_latest_build_url}/sha256sum-kernel" || return 1
-	else
-		sudo -E curl -fsOL "${latest_build_url}/sha256sum-kernel" || return 1
-        fi
+	sudo -E curl -fsOL "${latest_build_url}/sha256sum-kernel" || return 1
```

Comment:
Does this need to use `sudo`? We try to avoid running commands that hit the network as `root` unless absolutely essential.

Reply:
Maybe not.. note that's not a change in my patchset, though.  I'll look at adding another patch to the set which removes some unnecessary sudos.

Reply:
Looking closer we can't simply remove the `sudo` invocations: in each case the curl is writing to files in locations that the ordinary user won't have access to.  We could however `curl` without `sudo` to a temporary location then copy into the final location with `sudo`.  I'll look into implementing this.

Reply:
Ok, I've added a patch removing the `sudo curl`s to the series.